app/javascript/controllers/attachment_upload_controller.js

```diff
@@ -0,0 +1,68 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+    static targets = ["submitButton"];
+
+    connect() {
+        this.element.addEventListener("direct-upload:initialize", event => this.uploadInitialize(event))
+        this.element.addEventListener("direct-upload:start", event => this.uploadStart(event))
+        this.element.addEventListener("direct-upload:progress", event => this.uploadProgress(event))
+        this.element.addEventListener("direct-upload:error", event => this.uploadError(event))
+        this.element.addEventListener("direct-upload:end", event => this.uploadEnd(event))
+    }
+
+    uploadInitialize(event) {
+        const { target, detail } = event
+        const { id, file } = detail
+        target.insertAdjacentHTML("beforebegin", `
+            <div id="direct-upload-${id}"
+            class="direct-upload direct-upload--pending flex-1 relative mb-1">
+                <div class="flex justify-between mb-1 direct-upload__filename">
+                </div>
+                    <div class="w-full bg-gray-200 rounded-full h-2.5 dark:bg-gray-700">
+                    <div class="bg-blue-600 h-2.5 rounded-full" style="width: 0%" id="direct-upload-progress-${id}"></div>
+                </div>
+            </div>
+            `)
+        target.previousElementSibling.querySelector(`.direct-upload__filename`).innerHTML =
+            `<span class="text-base font-medium text-blue-700 dark:text-white">${file.name}</span>
+            <span class="text-sm font-medium text-blue-700 dark:text-white" id="upload-progress-${id}">0%</span>`
+
+        this.submitButtonTarget.disabled = true
+        this.submitButtonTarget.value = "Uploading"
```

Comment:
**nitpick:** Needs internationalization

Reply:
Just add it as a data attribute onto the element, you can do it like this https://stimulus.hotwired.dev/reference/values

Reply:
Fixed in da9120e5af5008763a1820a4528ef8b7e2fea73e

Reply:
Added translation to ef393daa27f8d30f1d8c73ee28fbf04263f9f63f